channelserver/src/server.rs

```diff
@@ -104,10 +112,12 @@ impl ChannelServer {
     ) -> Result<(), perror::HandlerError> {
         if let Some(participants) = self.channels.get_mut(channel) {
             // show's over, everyone go home.
-            if message == EOL {
+
+            if message.contains(&format!("\"message\":\"{}\"", EOL)) {
```

Comment:
Is the intention here that clients will deliberately send an empty message in order to terminate the connection?  If so, detecting it as a string match seems fragile.

Reply:
Not quite. This is going through `&format!` so what's actually generated is `"message":"^D"`. That's generated by the client sending a "^D". To be honest, I don't really expect there to be a strong case for clients sending a termination this way rather than just closing the link, but I've had experiences before where it's been desired. 

Totally fine dropping it if it's confusing.

Reply:
Aha, gotcha.  I think it's fine to leave in but probably worth a comment :-)